### PR TITLE
Make multiple datasets at once

### PR DESCRIPTION
**Pull request by RoriCremer**
This pr:
Uses the job ids of each create dataset job to set up a different preview page
Clears the dataset form once the create dataset job has been kicked off
Is the job id clearing needed?

3L6D9lB3Ux

### CODE REVIEW

File: src/components/DatasetCreateView.jsx [hunk 1]
```diff
@@ -176,6 +182,7 @@ export class DatasetCreateView extends React.PureComponent {
     const { asset, classes, ids, readers, studies, study } = this.props;
     const studyOptions = this.getStudyOptions(studies);
     const assetOptions = this.getAssetOptions(study);
+    const createDisabled = !study || !ids; // TODO move away from form for full validation
```
Comment (thathert) [inline src/components/DatasetCreateView.jsx]: How come?

File: src/sagas/repository.js [hunk 2]
```diff
@@ -88,13 +89,35 @@ function* pollJobWorker(jobId, jobTypeSuccess, jobTypeFailure) {
  * Datasets.
  */
 
-export function* createDataset({ payload }) {
+export function* createDataset() {
+  const dataset = yield select(getCreateDataset);
+  const payload = {
```
Comment (thathert) [inline src/sagas/repository.js]: Let's call this `datasetRequest` (I'm getting confused with the multiple uses of `payload` below.)

File: src/sagas/repository.js [hunk 3]
```diff
@@ -88,13 +89,35 @@ function* pollJobWorker(jobId, jobTypeSuccess, jobTypeFailure) {
...
+    ],
+  };
+  console.log(payload);
   try {
     const response = yield call(authPost, '/api/repository/v1/datasets', payload);
     const jobId = response.data.id;
     yield put({
       type: ActionTypes.CREATE_DATASET_JOB,
-      payload: { data: response, createdDataset: payload },
+      payload: { data: response, createdDataset: { jobId: payload } },
```
Comment (thathert) [inline src/sagas/repository.js]: You can't use a dynamic value like `jobId` as the key in the object here. It'd be nice to pull it out into the top level:
```javascript
{
    data: response,
    jobId: jobId,
    createdDataset: datasetRequest
}
```

File: src/sagas/repository.js [hunk 4]
```diff
@@ -88,13 +89,35 @@ function* pollJobWorker(jobId, jobTypeSuccess, jobTypeFailure) {
...
+      // The created dataset will now be defined by the job id
+      // I can't assign though,
+      // I need to add to the createdDataset? so that there is a lookup?
+      // when does the preview / getting created dataset state get cleared out?
```
Comment (thathert) [inline src/sagas/repository.js]: I'd say leave it in for now. It's not a lot of data per dataset request and I don't expect the initial users are going to be making thousands of datasets at a time so the chances of running out of memory are tiny.

File: src/sagas/repository.js [hunk 5]
```diff
@@ -88,13 +89,35 @@ function* pollJobWorker(jobId, jobTypeSuccess, jobTypeFailure) {
...
   try {
     const response = yield call(authPost, '/api/repository/v1/datasets', payload);
     const jobId = response.data.id;
     yield put({
       type: ActionTypes.CREATE_DATASET_JOB,
-      payload: { data: response, createdDataset: payload },
+      payload: { data: response, createdDataset: { jobId: payload } },
+      // The created dataset will now be defined by the job id
+      // I can't assign though,
+      // I need to add to the createdDataset? so that there is a lookup?
```
Comment (thathert) [inline src/sagas/repository.js]: Yes, I think we should do this in the reducer -- add something to the store for fetching datasetRequests by jobId.

**Review by thathert — APPROVED** [review]
LGTM

**Review by ahaessly — APPROVED** [review]
still just learning

**Pull request merged**